torch/csrc/autograd/python_function.cpp

```diff
@@ -201,6 +202,64 @@ auto PyNode::name() const -> std::string {
   return name;
 }
 
+#ifdef COMPILED_AUTOGRAD
+void PyNode::compiled_args(CompiledNodeArgs& args) {
+  static PyObject* method_name =
+      PyUnicode_InternFromString("_compiled_autograd_key");
+  THPObjectPtr pykey(PyObject_CallMethodNoArgs(obj, method_name));
+  if (!pykey)
+    throw_python_error();
+  TORCH_CHECK(
+      PyTuple_CheckExact(pykey.get()),
+      "_compiled_autograd_key shoud return tuple of ints");
+  auto size = PyTuple_GET_SIZE(pykey.get());
+  TORCH_CHECK(size > 0);
+  // first value is unique ID of the AotAutograd graph
+  ssize_t key = PyLong_AsSsize_t(PyTuple_GET_ITEM(pykey.get(), 0));
+  if (C10_UNLIKELY(key < 0)) {
+    TORCH_CHECK(PyErr_Occurred(), "key must be positive");
+    throw_python_error();
+  }
+  args.collect_size(static_cast<size_t>(key));
+  args.collect_size(size);
+
+  auto f = (THPFunction*)obj;
+  f->compiled_autograd_symints.clear();
+  f->compiled_autograd_symints.reserve(size - 1);
+  for (const auto i : c10::irange(1, size)) {
+    auto val = PyLong_AsSsize_t(PyTuple_GET_ITEM(pykey.get(), i));
+    if (C10_UNLIKELY(val == -1 && PyErr_Occurred()))
+      throw_python_error();
+    f->compiled_autograd_symints.emplace_back(val);
+  }
+
+  // AotAutograd symints are all dynamic
+  auto prior = args.set_default_dyn_type(SizeInput::DYNAMIC);
+  args.collect(f->compiled_autograd_symints);
+  args.set_default_dyn_type(prior);
+
+  args.collect(f->saved_variables);
+  args.collect(f->materialize_grads);
+  args.collect(f->is_variable_input);
+  // TODO(jansel): are there other things we need to collect?
```

Comment:
cc @albanD but I think you need to_save/non_differentiable/dirty_tensors as they are toggleable from Python and could vary from run to run. `materialize_non_diff_grads` is also new and needs to be added.

Reply:
Anything that would change the semantics and should invalidate the cache.